devp2p-eth/src/main/kotlin/org/apache/tuweni/devp2p/eth/EthRequestsManager.kt

```diff
@@ -30,75 +31,126 @@ import org.apache.tuweni.rlpx.wire.WireConnection
    * @param maxHeaders the max number of headers to provide
    * @param skip the headers to skip in between each header provided
    * @param reverse whether to provide headers in forward or backwards order
+   * @param connection the connection to use
    * @return a handle to the completion of the operation
    */
-  fun requestBlockHeaders(blockHash: Hash, maxHeaders: Long, skip: Long, reverse: Boolean): AsyncCompletion
+  fun requestBlockHeaders(
+    blockHash: Hash,
+    maxHeaders: Long,
+    skip: Long,
+    reverse: Boolean,
+    connection: WireConnection? = connectionSelectionStrategy().selectConnection()
+  ): AsyncResult<List<BlockHeader>>
+
   /**
    * Requests block headers
    * @param blockNumber the number of the block
    * @param maxHeaders the max number of headers to provide
    * @param skip the headers to skip in between each header provided
    * @param reverse whether to provide headers in forward or backwards order
+   * @param connection the connection to use
    * @return a handle to the completion of the operation
    */
-  fun requestBlockHeaders(blockNumber: Long, maxHeaders: Long, skip: Long, reverse: Boolean): AsyncCompletion
+  fun requestBlockHeaders(
+    blockNumber: Long,
+    maxHeaders: Long,
+    skip: Long,
+    reverse: Boolean,
+    connection: WireConnection? = connectionSelectionStrategy().selectConnection()
+  ): AsyncResult<List<BlockHeader>>
+
   /**
    * Requests block bodies from block hashes
    * @param blockHashes the hashes of the blocks
+   * @param connection the connection to use
    * @return a handle to the completion of the operation
    */
-  fun requestBlockBodies(blockHashes: List<Hash>): AsyncCompletion
+  fun requestBlockBodies(
+    blockHashes: List<Hash>,
+    connection: WireConnection? = connectionSelectionStrategy().selectConnection()
+  ): AsyncResult<List<BlockBody>>
+
   /**
    * Requests a block from block hash
    * @param blockHash the hash of the block
+   * @param connection the connection to use
    * @return a handle to the completion of the operation
    */
-  fun requestBlock(blockHash: Hash): AsyncCompletion
+  fun requestBlock(
+    blockHash: Hash,
+    connection: WireConnection? = connectionSelectionStrategy().selectConnection()
+  ): AsyncResult<Block>
+
   /**
    * Requests transaction receipts
    * @param blockHashes the hashes to request transaction receipts for
+   * @param connection the connection to use
    * @return a handle to the completion of the operation
    */
-  fun requestTransactionReceipts(blockHashes: List<Hash>): AsyncCompletion
+  fun requestTransactionReceipts(
+    blockHashes: List<Hash>,
+    connection: WireConnection? = connectionSelectionStrategy().selectConnection()
+  ): AsyncResult<List<List<TransactionReceipt>>>
+
   /**
    * Checks if a request was made to get block headers
    * @param connection the wire connection sending data
    * @param header the block header just received
```

Comment:
should be  "@param headers" (plural) and i don't know if the sentence should be "@params headers list of block headers just received" or something more specific